stdlib/2and3/select.pyi

```diff
@@ -73,63 +24,116 @@ class poll:
-    def unregister(self, fd: FileDescriptorLike) -> None: ...
-    def poll(self, timeout: float = ..., maxevents: int = ...) -> List[Tuple[int, int]]: ...
-    @classmethod
-    def fromfd(cls, fd: FileDescriptorLike) -> epoll: ...
+    KQ_EV_ADD: int
+    KQ_EV_CLEAR: int
+    KQ_EV_DELETE: int
+    KQ_EV_DISABLE: int
+    KQ_EV_ENABLE: int
+    KQ_EV_EOF: int
+    KQ_EV_ERROR: int
+    KQ_EV_FLAG1: int
+    KQ_EV_ONESHOT: int
+    KQ_EV_SYSFLAGS: int
+    KQ_FILTER_AIO: int
+    KQ_FILTER_NETDEV: int
+    KQ_FILTER_PROC: int
+    KQ_FILTER_READ: int
+    KQ_FILTER_SIGNAL: int
+    KQ_FILTER_TIMER: int
+    KQ_FILTER_VNODE: int
+    KQ_FILTER_WRITE: int
+    KQ_NOTE_ATTRIB: int
+    KQ_NOTE_CHILD: int
+    KQ_NOTE_DELETE: int
+    KQ_NOTE_EXEC: int
+    KQ_NOTE_EXIT: int
+    KQ_NOTE_EXTEND: int
+    KQ_NOTE_FORK: int
+    KQ_NOTE_LINK: int
+    if sys.platform != "darwin":
+        KQ_NOTE_LINKDOWN: int
+        KQ_NOTE_LINKINV: int
+        KQ_NOTE_LINKUP: int
+    KQ_NOTE_LOWAT: int
+    KQ_NOTE_PCTRLMASK: int
+    KQ_NOTE_PDATAMASK: int
+    KQ_NOTE_RENAME: int
+    KQ_NOTE_REVOKE: int
+    KQ_NOTE_TRACK: int
+    KQ_NOTE_TRACKERR: int
+    KQ_NOTE_WRITE: int
 
-if sys.version_info >= (3, 3):
-    # Solaris only
-    class devpoll:
+if sys.platform == "linux":
+    class epoll(object):
+        if sys.version_info >= (3, 3):
+            def __init__(self, sizehint: int = ..., flags: int = ...) -> None: ...
+        else:
+            def __init__(self, sizehint: int = ...) -> None: ...
         if sys.version_info >= (3, 4):
-            def close(self) -> None: ...
-            closed: bool
-            def fileno(self) -> int: ...
+            def __enter__(self) -> epoll: ...
+            def __exit__(self, exc_type: Optional[Type[BaseException]] = ..., exc_val: Optional[BaseException] = ..., exc_tb: Optional[TracebackType] = ...) -> None: ...
+        def close(self) -> None: ...
+        closed: bool
+        def fileno(self) -> int: ...
         def register(self, fd: FileDescriptorLike, eventmask: int = ...) -> None: ...
-        def modify(self, fd: FileDescriptorLike, eventmask: int = ...) -> None: ...
+        def modify(self, fd: FileDescriptorLike, eventmask: int) -> None: ...
         def unregister(self, fd: FileDescriptorLike) -> None: ...
-        def poll(self, timeout: Optional[float] = ...) -> List[Tuple[int, int]]: ...
+        def poll(self, timeout: Optional[float] = ..., maxevents: int = ...) -> List[Tuple[int, int]]: ...
+        @classmethod
+        def fromfd(cls, __fd: FileDescriptorLike) -> epoll: ...
+
+    EPOLLERR: int
+    EPOLLET: int
+    EPOLLHUP: int
+    EPOLLIN: int
+    EPOLLMSG: int
+    EPOLLONESHOT: int
+    EPOLLOUT: int
+    EPOLLPRI: int
+    EPOLLRDBAND: int
+    EPOLLRDNORM: int
+    EPOLLWRBAND: int
+    EPOLLWRNORM: int
+    EPOLL_RDHUP: int
+
+if sys.platform != "linux" and sys.platform != "darwin" and sys.platform != "win32":
```

Comment:
A quick google seems to suggest that Solaris is identified as "sunos", but I am not totally confident about this.

Reply:
yes, it seems that it would be sunosX, where X is some numeral. unfortunately flake8-pyi (and I think pytype) don’t allow startswith checks. flake8-pyi also doesn’t permit sunos as the constant. 
similar situation applies for cases where we could type bsd.

Reply:
It's pretty easy to add more allowed platforms to flake8-pyi—I think I just added all the ones that existed in typeshed when I wrote the check. The intent is to protect against typos in the platform name.

Reply:
yeah, the bigger issue is the `startswith`, which requires changes to both flake8-pyi and pytype.